File: 프로그래머스/unrated/176963. 추억 점수/추억 점수.js
function solution(name, yearning, photo) {
    var answer = [];
    let obj = new Map();
    for(let i =0; i< name.length; i++){
        obj.set(name[i],yearning[i])
    }
    answer =photo.map((v) => v.map(i => 
                                   obj.has(i)?obj.get(i):0).reduce((acc, curr) => acc+curr))
    return answer;
}

File: 프로그래머스/unrated/176963. 추억 점수/추억 점수.py
def solution(name, yearning, photo):
    answer = []
    yearning_dict = dict(zip(name,yearning))

    for p in photo:
        score=0
        for name in p:
            if name in yearning_dict:
                score += yearning_dict[name]
        answer.append(score)
    return answer